gamedata/alldefs_post.lua

```diff
@@ -1373,6 +1373,55 @@ function WeaponDef_Post(name, wDef)
 			end
 		end
 
+		if modOptions.shieldsrework == true then
+			local shieldCollisionExemptions = { --add the name of the weapons (or just the name of the unit followed by _ ) to this table to exempt from shield collision. 
+				'corsilo_', 'armsilo_', 'armthor_empmissile', 'armemp_', 'cortron_', 'corjuno_', 'armjuno_'
+			}
+
+			-- For balance, paralyzers need to do reduced damage to shields, as their raw raw damage is outsized
+			local paralyzerShieldDamageMultiplier = 0.25
+
+			if wDef.shield then
+				wDef.shield.repulser = false
+				wDef.shield.exterior = true
+			end
+
+			if (not wDef.interceptedbyshieldtype) or wDef.interceptedbyshieldtype ~= 1 then
+				for _, exemption in ipairs(shieldCollisionExemptions) do
+					if string.find(name, exemption)then
+						wDef.interceptedbyshieldtype = 4
+						break
+					else
+						wDef.interceptedbyshieldtype = 1
+					end
+				end
+			end
+
+			if wDef.damage ~= nil and wDef.damage.default ~= nil then
+				wDef.customparams = wDef.customparams or {}
+				-- Due to the engone not handling overkill damage, we have to store the original shield damage values as a customParam for unit_shield_behavior.lua to reference
+				wDef.customparams.shield_damage = wDef.damage.shields or wDef.damage.default
+
+				if wDef.paralyzer then
+					wDef.customparams.shield_damage = wDef.customparams.shield_damage * paralyzerShieldDamageMultiplier
+				end
+
+				-- Set damage to 0 so projectiles always collide with shield. Without this, if damage > shield charge then it passes through.
+				-- Applying damage is instead handled in unit_shield_behavior.lua
+				wDef.damage.shields = 0
+
+				if wDef.beamtime and wDef.beamtime > 0.0333334 then --0.0333334 is the minimum beamtime required to register a single frame hit on ShieldPreDamaged() callin
+					 -- This splits up the damage of hitscan weapons over the duration of beamtime, as each frame counts as a hit in ShieldPreDamaged() callin
+					 -- Math.floor is used to sheer off the extra digits of the number of frames that the hits occur
+					wDef.customparams.beamtime_damage_reduction_multiplier = (1 / math.floor((wDef.beamtime * 30)))
```

Comment:
magic constant
```suggestion
					wDef.customparams.beamtime_damage_reduction_multiplier = (1 / math.floor((wDef.beamtime * Game.gameSpeed)))
```

Reply:
`Game.gameSpeed` is the simulation speed, in frames per second. It is _not_ related to the player adjustable game speed from x0.25 - x20.

Reply:
Ahh, I understand now. Thanks gentlemen